protocol/chainlib/jsonRPC.go

```diff
@@ -484,6 +484,11 @@ func (cp *JrpcChainProxy) SendNodeMsg(ctx context.Context, ch chan interface{},
 			return nil, "", nil, utils.LavaFormatError("jsonRPC error", err, utils.Attribute{Key: "GUID", Value: ctx})
 		}
 		replyMsg = *replyMessage
+
+		err := cp.HandleIDMismatchError(nodeMessage.ID, replyMessage.ID)
```

Comment:
lets change the name of this to ValidateRequestAndResponseIds 
so its clear its a validation.

Reply:
fixed: 3e029e715b1a279a51f3bf50120ead79aaa4cffe